### DISCUSSION TITLE
Save bank with midi routing

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#59] [Q&A]
Hi everyone,
I'd like to save patch inside a bank with midi routing rules (I've connected my alesis vortex and my tyros 5 to squishbox and sometimes i want to use tyros sound with wireless keytar using midi thru or something like that)
Thanks for the help.

**Comment by albedozero** [comment]
I'm a little confused about what you're asking - can you be more specific? Do you want to use the "Save Patch" option on the SquishBox to save the current patch as a new patch, with the same instruments and router rules? This is what should be happening by default.

**Reply by tizfra** [reply]
Ok, 
I want to save how I "connected" midi device in system, midi device.
Because I use midi thru to connect alesis vortex to tyros.
Thanks

**Reply by albedozero** [reply]
I see now, thanks. I think the best way to do this is the "MIDI way" like @garamoniker says. Sound sources should be on different MIDI channels, and controllers (i.e. the Vortex) should be able to switch between MIDI channels to trigger specific sound sources.

**Comment by garamoniker** [comment]
I don't know if this is too primitive a fix for you but why don't you set up your rig something like this.

Vortex -  Tyros on Midi Channel 1
               Squishbox on Channel 2

I know on my Vortex I can easily switch between midi channels.

You could keep your "midi" version of the Tyros on Midi Channel 2 as well. I'm not familiar with the Tyros but I assume you can set up the midi however you want? Then, maybe, you'd just have to remember to turn down the Tyros' sound volume when you want to use it as a controller for the Squishbox.
